docs/api_information/retries_and_idempotency.md

```diff
@@ -0,0 +1,15 @@
+<h3>Retries &amp; Idempotency</h3>
+
+<b>Process Authorisation</b>
+
+The very first thing that's done on calls to the ProcessAuthorisation API, is that the pre-approval code (as specified by the x_pre_approval_code parameter) is flagged as 'used'. Once used, a pre-approval code can never again be reused to get consumer finance. If an already used pre-approval code is passed through on a call to ProcessAuthorisation, it will not be re-processed - the current authorisation status will instead be returned.
```

Comment:
Idempotency is driven off x_pos_transaction_ref and not barcode .. However, we do have a story to include barcode in nonce calculation. The very first thing the API does is to lookup "x_pos_transaction_ref " (for the specific merchant) in our DB to see if we have already received this nonce before. If yes, we return the previous result else we process it.

In case of retries, POS can retry with the same payload (including same pos_transaction_ref) and we will return the original response, without processing it again.

hope it makes sense